bundles/org.pitest.pitclipse.launch.ui/src/org/pitest/pitclipse/launch/ui/PitArgumentsTab.java

```diff
@@ -38,37 +54,79 @@
 import org.pitest.pitclipse.core.PitCoreActivator;
 import org.pitest.pitclipse.runner.config.PitConfiguration;
 
-import static org.eclipse.jdt.launching.IJavaLaunchConfigurationConstants.ATTR_MAIN_TYPE_NAME;
-import static org.eclipse.jdt.launching.IJavaLaunchConfigurationConstants.ATTR_PROJECT_NAME;
-import static org.eclipse.swt.layout.GridData.FILL_HORIZONTAL;
-import static org.pitest.pitclipse.core.preferences.PitPreferences.AVOID_CALLS_FROM_PIT;
-import static org.pitest.pitclipse.core.preferences.PitPreferences.EXCLUDE_CLASSES_FROM_PIT;
-import static org.pitest.pitclipse.core.preferences.PitPreferences.EXCLUDE_METHODS_FROM_PIT;
-import static org.pitest.pitclipse.core.preferences.PitPreferences.MUTATION_TESTS_RUN_IN_PARALLEL;
-import static org.pitest.pitclipse.core.preferences.PitPreferences.USE_INCREMENTAL_ANALYSIS;
-import static org.pitest.pitclipse.launch.config.LaunchConfigurationWrapper.ATTR_AVOID_CALLS_TO;
-import static org.pitest.pitclipse.launch.config.LaunchConfigurationWrapper.ATTR_EXCLUDE_CLASSES;
-import static org.pitest.pitclipse.launch.config.LaunchConfigurationWrapper.ATTR_EXCLUDE_METHODS;
-import static org.pitest.pitclipse.launch.config.LaunchConfigurationWrapper.ATTR_TEST_INCREMENTALLY;
-import static org.pitest.pitclipse.launch.config.LaunchConfigurationWrapper.ATTR_TEST_IN_PARALLEL;
-import static org.pitest.pitclipse.launch.PitLaunchArgumentsConstants.ATTR_TEST_CONTAINER;
-
 /**
  * Tab allowing to configure a PIT analyze. 
  */
 public final class PitArgumentsTab extends AbstractLaunchConfigurationTab {
     private static final int NUMBER_OF_COLUMNS = 3;
 
-    private Text testClassText;
+    /**
+     * Text which holds the information about which project to mutate.
+     */
     private Text projectText;
+    private String containerId;
+    /**
+     * Radio button, if selected a test class is specified with
+     * {@link #testClassText}.
+     */
     private Button testClassRadioButton;
+    /**
+     * Text which holds the information about which test class is used
+     */
+    private Text testClassText;
+    /**
+     * Radio button, if selected a test directory is specified with
+     * {@link #testDirText}.
+     */
     private Button testDirectoryRadioButton;
+    /**
+     * Text which holds the information about which directory the tests are located
+     */
     private Text testDirText;
-    private String containerId;
+    /**
+     * Check box button, if selected a single or more target class is specified with
+     * {@link #targetClassText}.
+     */
+    private Button targetClassCheckBoxButton;
+    /**
+     * Text which holds the information about which classes should be mutated
+     */
+    private Text targetClassText;
+    /**
+     * Pattern which matches a String which contains packages separated by dots and
+     * end with a class.
+     */
+    private static final String CLASS_PATTERN = "\\w+(\\.\\w+)*\\.\\w+";
```

Comment:
Please keep in mind that `$` is a valid character in a class name, while they cannot start with a number...
maybe JDT itself already provides some validation under that respect?

Reply:
Actually it's handled by Java itself, please have a look at https://stackoverflow.com/questions/5205339/regular-expression-matching-fully-qualified-class-names 

`Pattern.compile("\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*\\.\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*)*");`

Reply:
Looks good, thank you.